MANUAL.md

```diff
@@ -0,0 +1,194 @@
+# Testkit manual
+
+Testkit for Exonum blockchain framework, allowing to test service APIs synchronously and in the same process as the testkit.
+
+Below is a simple user guide.
+
+* [Installation](#installation)
+* [Transactions testing](#transactions-testing)
+* [Api testing](#api-testing)
+* [Oracles testing](#oracles-testing)
+* [Configuration changes testing](#configuration-changes-testing)
+
+## Installation
+
+Just add a following line to the `Cargo.toml`:
+
+```toml
+[dev-dependencies]
+exonum-testkit = "0.1.0"
+```
+
+## Simple usage
+
+### Transactions testing
+
+The primary goal of the kind of tests is to check the business logic of your service.
+
+For writing your first test create `tests` directory in according to the cargo
+integration testing [manual][integration-tests].
+After that creates file `transactions.rs` like the following content.
+
+```rust
+extern crate exonum;
+extern crate exonum_testkit;
+extern crate my_service;
+
+use my_service::{MyService, MyTransaction, MySchema};
+use exonum_testkit::TestKitBuilder;
+
+#[test]
+fn test_my_tx() {
+    // Create simple testkit network.
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(MyService::new())
+        .create();
+    // Create transaction.
+    let tx = MyTransaction::new(...);
+    // Commit it into blockchain.
+    testkit.create_block_with_transactions(txvec![tx.clone()]);
+    // Check the expected result
+    let snapshot = testkit.snapshot();
+    let schema = MySchema::new(&snapshot);
+    assert!(schema.is_my_data_checked());
+}
+```
+
+Make sure that you have full coverage of the business logic in the `execute` method of your transactions.
+But if you just want to check the `verify` logic in the transaction, you can do it without testkit in
+a simple way like this:
+
+```rust
+let tx = MyTransaction::new(...);
+assert!(tx.verify());
+```
+
+### Api testing
+
+The following steps may help you.
+
+* Define the `MyServiceApi` trait for the `TestKitApi` structure that covers the whole api of your service.
+* Implement functions that use transactions to fill your storage with the test data.
```

Comment:
Nitpick: could you detail on this a little bit, please? If you suggest to write to the storage bypassing transactions, I'm not sure it's the best way to do things, because it could break invariants checked by `execute()` method in transactions. OTOH, it could be inevitable in some cases...

Reply:
I suggest to write data to the storage during the transaction execute. But perhaps did not accurately formulate it.